rust/cadeau/src/xmf/vpx/mod.rs

```diff
@@ -0,0 +1,225 @@
+use core::fmt;
+
+use xmf_sys::{
+    vpx::{XmfVpxCodecType, XmfVpxPacketKind},
+    XmfVpxFrame, XmfVpxFrame_Destroy, XmfVpxFrame_GetBuffer, XmfVpxFrame_GetDuration, XmfVpxFrame_GetFlags,
+    XmfVpxFrame_GetHeight, XmfVpxFrame_GetPartitionId, XmfVpxFrame_GetPts, XmfVpxFrame_GetSize,
+    XmfVpxFrame_GetSpatialLayerEncoded, XmfVpxFrame_GetWidth, XmfVpxImage, XmfVpxImage_Destroy, XmfVpxPacket,
+    XmfVpxPacket_Destroy, XmfVpxPacket_GetFrame, XmfVpxPacket_GetKind, XmfVpxPacket_IsEmpty,
+};
+
+pub mod decoder;
+pub mod encoder;
+
+#[derive(Debug, Clone, Copy)]
+pub enum VpxCodec {
+    VP8,
+    VP9,
+}
+
+pub struct VpxImage {
+    pub(crate) ptr: *const XmfVpxImage,
+}
+
+impl Drop for VpxImage {
+    fn drop(&mut self) {
+        // Safety: it is safe to call, the owenership of the pointer is managed by the XmfVpxImage itself.
+        unsafe {
+            XmfVpxImage_Destroy(self.ptr.cast_mut());
```

Comment:
issue: `cast_mut` is generally not a good idea. Try to avoid that. In this case, why `ptr` is a `*const XmfVpxImage`? `XmfVpxDecoder_GetNextFrame` is returning `*mut XmfVpxImage`. Use this directly. Mut-to-const is fine, but the other way around is problematic if the original pointer was not already mut, and I would rather avoid tracking this information in my head.

Reply:
updated